pkg/kubectl/cmd/apply_test.go

```diff
@@ -990,6 +998,91 @@ func TestRunApplySetLastApplied(t *testing.T) {
 	cmdutil.BehaviorOnFatal(func(str string, code int) {})
 }
 
+func TestRunApplyDiffLastApplied(t *testing.T) {
+	initTestErrorHandler(t)
+	rcByte := readBytesFromFile(t, filenameApplyDiffRESP)
+	pathRC := "/namespaces/test/replicationcontrollers/" + "test-rc"
+
+	rcChangedByte := readBytesFromFile(t, filenameApplyDiffChangedRESP)
+
+	tests := []struct {
+		name, nameRC, pathRC, filePath, expectedGitDiffOut, expectedDiffOut, expectedErr, output string
+		resp                                                                                     []byte
+	}{
+		{
+			name:               "git-diff objects with no different",
+			filePath:           filenameRC,
+			expectedErr:        "",
+			expectedGitDiffOut: "",
+			expectedDiffOut:    "",
+			resp:               rcByte,
+			output:             "yaml",
+		},
+		{
+			name:               "git-diff after local file changed with json",
+			filePath:           filenameLocalChangedRC,
+			expectedErr:        "",
+			expectedGitDiffOut: "-                \"containerPort\": 123\n+                \"containerPort\": 80\n               }\n",
+			expectedDiffOut:    "<                 \"containerPort\": 123\n>                 \"containerPort\": 80\n",
+			resp:               rcByte,
+			output:             "json",
+		},
+		{
+			name:               "git-diff after annotations changed",
+			filePath:           filenameRC,
+			expectedErr:        "",
+			expectedGitDiffOut: "-        - containerPort: 80\n+        - containerPort: 123\n",
+			expectedDiffOut:    "<         - containerPort: 80\n>         - containerPort: 123\n",
+			resp:               rcChangedByte,
+			output:             "yaml",
+		},
+	}
+	for _, test := range tests {
+		f, tf, _, _ := cmdtesting.NewAPIFactory()
+		tf.Printer = &testPrinter{}
+		tf.UnstructuredClient = &fake.RESTClient{
+			APIRegistry:          api.Registry,
+			NegotiatedSerializer: unstructuredSerializer,
+			Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+				switch p, m := req.URL.Path, req.Method; {
+				case p == pathRC && m == "GET":
+					bodyRC := ioutil.NopCloser(bytes.NewReader(test.resp))
+					return &http.Response{StatusCode: 200, Header: defaultHeader(), Body: bodyRC}, nil
+				default:
+					t.Fatalf("unexpected request: %#v\n%#v", req.URL, req)
+					return nil, nil
+				}
+			}),
+		}
+		tf.Namespace = "test"
+		tf.ClientConfig = defaultClientConfig()
+		buf, errBuf := bytes.NewBuffer([]byte{}), bytes.NewBuffer([]byte{})
```

Comment:
from the [Go documentation](https://golang.org/pkg/bytes/#NewBuffer):
> In most cases, new(Buffer) (or just declaring a Buffer variable) is sufficient to initialize a Buffer.

There is no behavior difference. I have a personal preference for the most simple or common pattern, which in this case is new(Buffer). Your call.